pkg/sfsturbo/services/volumes.go

```diff
@@ -0,0 +1,174 @@
+package services
+
+import (
+	"strconv"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/sfs_turbo/v1/shares"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	log "k8s.io/klog/v2"
+)
+
+const (
+	shareCreating  = "100"
+	shareAvailable = "200"
+	shareDeleted   = "400"
+
+	shareSubExpanding     = "121"
+	shareSubExpandSuccess = "221"
+	shareSubExpandError   = "321"
+
+	shareDescription = "provisioned-by=sfsturbo.csi.huaweicloud.org"
+)
+
+func CreateShareCompleted(client *golangsdk.ServiceClient, createOpts *shares.CreateOpts) (
+	*shares.TurboResponse, error) {
+	turboResponse, err := CreateShare(client, createOpts)
+	if err != nil {
+		return nil, err
+	}
+	log.V(4).Infof("[DEBUG] create share response detail: %v", turboResponse)
+	err = WaitForShareAvailable(client, turboResponse.ID)
+	if err != nil {
+		return nil, err
+	}
+	return turboResponse, nil
+}
+
+// WaitForShareAvailable create new share from sfsturbo
+func WaitForShareAvailable(client *golangsdk.ServiceClient, shareID string) error {
+	return common.WaitForCompleted(func() (bool, error) {
+		share, err := GetShare(client, shareID)
+		if err != nil {
+			return false, status.Errorf(codes.Internal,
+				"Failed to query share %s when wait share available: %v", shareID, err)
+		}
+		log.V(4).Infof("[DEBUG] query share detail: %v", share)
+		if share.Status == shareAvailable {
+			return true, nil
+		}
+		if share.Status == shareCreating {
+			return false, nil
+		}
+		return false, status.Error(codes.Internal, "created share status is not available")

```

Comment:
how about the following:
`status.Errorf(codes.Internal, "created share status is not available : %s", share.Status)`

Reply:
done